### ISSUE TITLE
Provide better error message (or more specific to situation) for evaluation order issues.

### ISSUE THREAD

**Issue by pbennett** [tinymanorg/tealish#54]
A statement like:
if a == 2 && b ==3:
will yield an error like:
Error: Cannot parse "if a == 2 && b ==3" as Expression at line 100

I had a more  involved if statement in my particular case but I fought for quite a bit of time trying to figure out what was wrong (and there's no examples in the repo of an && expression like above) until I realized it wanted
if (a==2) && (b==3):

Tealish should make it extremely clear that the evaluation order isn't defined and needs to be.  The error instead reads almost like a generic 'something's wrong' - ie: a syntax error.

**Comment by fergalwalsh** [comment]
Yep, this is definitely an area that needs attention. I do want the errors to be very precise and useful but it will take a bit of work to get there. Definitely in the plan though.
